test/run-test.ps1

```diff
@@ -1,169 +1,62 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
 [cmdletbinding()]
 param(
-    [switch]$UseImageCache,
+    [string]$DotnetInstallDir,
     [string]$Filter,
     [string]$Architecture
 )
 
-Set-StrictMode -Version Latest
-$ErrorActionPreference = 'Stop'
-
-function Exec([scriptblock]$cmd, [string]$errorMessage = "Error executing command: " + $cmd) {
-    & $cmd
-    if ($LastExitCode -ne 0) {
-        throw $errorMessage
-    }
-}
-
-function Get-ActivePlatformImages([PSCustomObject]$manifestRepo, [string]$activeOS) {
-    return $manifestRepo.Images |
-        ForEach-Object { $_.Platforms } |
-        Where-Object { $_.os -eq "$activeOS" } |
-        Where-Object { ( [string]::IsNullOrEmpty($Architecture) -and -not [bool]($_.PSobject.Properties.name -match "architecture"))`
-            -or ( [bool]($_.PSobject.Properties.name -match "architecture") -and $_.architecture -eq "$Architecture" ) }
-}
+$DotnetInstallDir = "$PSScriptRoot/../.dotnet"
 
-function Get-RuntimeTag([string]$sdkDockerfilePath, [string]$runtimeType, [string]$activeOS, [PSCustomObject]$manifestRepo) {
-    $runtimeDockerfilePath = $sdkDockerfilePath.Replace("sdk", $runtimeType)
-    $platforms = Get-ActivePlatformImages $manifestRepo $activeOS |
-        Where-Object { $_.Dockerfile -eq $runtimeDockerfilePath }
-    return $manifestRepo.Name + ':' + ([array]($_.Tags | ForEach-Object { $_.PSobject.Properties }))[0].Name
+if (!(Test-Path "$DotnetInstallDir")) {
+    mkdir "$DotnetInstallDir" | Out-Null
 }
 
-if ($UseImageCache) {
-    $optionalDockerBuildArgs = ""
+# Install the .NET Core SDK
+$IsRunningOnUnix = $PSVersionTable.contains("Platform") -and $PSVersionTable.Platform -eq "Unix"
+if ($IsRunningOnUnix) {
+    $DotnetInstallScript = "dotnet-install.sh"
 }
 else {
-    $optionalDockerBuildArgs = "--no-cache"
+    $DotnetInstallScript = "dotnet-install.ps1"
 }
 
-$dirSeparator = [IO.Path]::DirectorySeparatorChar
-$repoRoot = Split-Path -Parent $PSScriptRoot
-$manifestPath = [IO.Path]::combine(${repoRoot}, "manifest.json")
-$manifestRepo = (Get-Content $manifestPath | ConvertFrom-Json).Repos[0]
-$testFilesPath = "$PSScriptRoot$dirSeparator"
-$activeOS = docker version -f "{{ .Server.Os }}"
-
-# update as appropriate (e.g. "2.0-sdk") whenever pre-release packages are referenced prior to being available on NuGet.org.
-$includePrereleasePackageSourceForSdkTag = "2.*-sdk*"
+if (!(Test-Path $DotnetInstallScript)) {
+    $DOTNET_INSTALL_SCRIPT_URL = "https://raw.githubusercontent.com/dotnet/cli/release/2.0.0/scripts/obtain/$DotnetInstallScript"
+    Invoke-WebRequest $DOTNET_INSTALL_SCRIPT_URL -OutFile $DotnetInstallDir/$DotnetInstallScript
```

Comment:
Should we check `$LASTEXITCODE` to ensure download succeeded?

Reply:
I think `Invoke-WebRequest` would be a try-catch instead of `$LASTEXITCODE` since it's internal to PowerShell.

Reply:
No - This is a cmdlet.  I will set $ErrorActionPreference to stop to handle this.

Reply:
Thanks.  Couple follow up questions -

1. Do we want to wait-retry?
2. Should we get this from Azure and not GitHub?

Reply:
1.  In the long run yes - I am going to be doing some additional refactoring soon in which I will address this.
2.  I don't see a reasonable option for this as CLI doesn't consider their install script a build artifact therefore it is only available via GitHub.